neuraxle/hyperparams/scipy_distributions.py

```diff
@@ -0,0 +1,1010 @@
+
+    @scipy_method
+    def std(self, *args, **kwargs):
+        """
+        Standard deviation of the distribution.
+
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        return self.scipy_distribution.std(*args, **kwargs, **self.scipy_distribution_arguments)
+
+    @scipy_method
+    def support(self, *args, **kwargs):
+        """
+        Return the support of the distribution.
+
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        return self.scipy_distribution.support(*args, **kwargs, **self.scipy_distribution_arguments)
+
+    @scipy_method
+    def var(self, *args, **kwargs):
+        """
+        Variance of the distribution.
+
+        :return:
+        """
+        return self.scipy_distribution.var(*args, **kwargs, **self.scipy_distribution_arguments)
+
+    def to_sk_learn(self):
+        return self.scipy_distribution
+
+
+class BaseDiscreteDistribution(ScipyDistributionWrapper):
+    def __init__(self, name, min_included, max_included, null_default_value, **kwargs):
+        scipy_dist_obj = rv_discrete(
+            name=name,
+            a=min_included,
+            b=max_included,
+            **kwargs
+        )
+        scipy_dist_obj._pmf = self._pmf
+        super().__init__(
+            scipy_distribution=scipy_dist_obj,
+            null_default_value=null_default_value,
+            is_continuous=False
+        )
+
+    def _override_scipy_methods(self):
+        self.scipy_distribution._pmf = self._pmf
+
+    @abstractmethod
+    def _pmf(self, x, *args):
+        pass
+
+
+class Distribution(rv_continuous):
+    def _pdf(self, x):
+        return 0.0
+
+
+class BaseContinuousDistribution(ScipyDistributionWrapper):
+    def __init__(self, name, min_included, max_included, null_default_value, **kwargs):
+        scipy_distribution = Distribution(
+            name=name,
+            a=min_included,
+            b=max_included,
+            **kwargs
+        )
+        scipy_distribution._pdf = self._pdf
+
+        super().__init__(
+            scipy_distribution=scipy_distribution,
+            null_default_value=null_default_value,
+            is_continuous=True
+        )
+
+    def _override_scipy_methods(self):
+        self.scipy_distribution._pdf = self._pdf
+
+    @abstractmethod
+    def _pdf(self, x, *args):
+        pass
+
+
+
+class RandInt(ScipyDistributionWrapper):
+    """
+    Rand int scipy distribution. Check out `scipy.stats.randint for more info <https://docs.scipy.org/doc/scipy-0.15.1/reference/generated/scipy.stats.randint.html>`_
+    """
+
+    def __init__(self, min_included: int, max_included: int, null_default_value: float = None, **kwargs):
+        self.min_included = min_included
+        self.max_included = max_included
+        kwargs.update(low=min_included, high=max_included)
+
+        super().__init__(
+            scipy_distribution=randint,
```

Comment:
I would have been expecting the ability of a user to do this: 

`ScipyDistributionWrapper(randint(low=2, high=5))` 

Such that the boilerplate be minimal like this: 

```
hyperparam_space = {
    'lol': randint(low=2, high=5),  # scipy
    'hey': RandInt(2, 5)  # neuraxle
}

p.update_hyperparams_space(hyperparam_space)

_subtype = "rvdiscrete"  # or else use "ContinuousDistribution" of Neuraxle, but the same subtype must match for both the "lol" and "hey" at the same type. 
assert isinstance(p.get_hyperparams(['lol'], _subtype)
assert isinstance(p.get_hyperparams(['hey'], _subtype)

```

This way the `p.update_hyperparams_space` would itself use the `ScipyDistributionWrapper` in the base step.

This is what I was meaning by being compatible with scipy. 

I'm sorry the present file was rather long to write before validating the concept.

Reply:
you can do this : 
`
ScipyDistributionWrapper(
     scipy_distribution=randint(low=0, high=10),
     is_continuous=False,
     null_default_value=0
  )
`

Reply:
the last 2 parameters could be removed also